dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java

```diff
@@ -660,34 +665,24 @@ private void checkToolTokens(User authUser, Long userId, String organization) {
     public List<EntryUpdateTime> getUserEntries(@ApiParam(hidden = true) @Parameter(hidden = true, name = "user", in = ParameterIn.HEADER) @Auth User authUser,
                                                 @Parameter(name = "count", description = "Maximum number of entries to return", in = ParameterIn.QUERY) @QueryParam("count") Integer count,
                                                 @Parameter(name = "filter", description = "Filter paths with matching text", in = ParameterIn.QUERY) @QueryParam("filter") String filter) {
-        final List<EntryUpdateTime> entryUpdateTimes = new ArrayList<>();
-        final User fetchedUser = this.userDAO.findById(authUser.getId());
-
-        Set<Entry> entries = fetchedUser.getEntries();
-        entries.forEach(entry -> {
-            Timestamp timestamp = entry.getDbUpdateDate();
-            Set<Version> versions = entry.getWorkflowVersions();
-            Optional<Version> mostRecentTag = versions.stream().max(Comparator.comparing(Version::getDbUpdateDate));
-            if (mostRecentTag.isPresent() && timestamp.before(mostRecentTag.get().getDbUpdateDate())) {
-                timestamp = mostRecentTag.get().getDbUpdateDate();
-            }
-            List<String> pathElements = Arrays.asList(entry.getEntryPath().split("/"));
-            String prettyPath = String.join("/", pathElements.subList(2, pathElements.size()));
-            entryUpdateTimes.add(new EntryUpdateTime(entry.getEntryPath(), prettyPath, entry.getEntryType(), timestamp));
-        });
-
-        // Sort all entryUpdateTimes by timestamp
-        List<EntryUpdateTime> sortedEntries = entryUpdateTimes
-                .stream()
+        //get entries with only minimal columns from database
+        final List<EntryLite> entriesLite = new ArrayList<>();
+        final long userId = authUser.getId();
+        entriesLite.addAll(toolDAO.findEntryVersions(userId));
+        entriesLite.addAll(bioWorkflowDAO.findEntryVersions(userId));
+        entriesLite.addAll(serviceDAO.findEntryVersions(userId));
+
+        //cleanup fields for UI, filter, and sort
+        List<EntryUpdateTime> filteredEntries = entriesLite
+                .stream().map(e -> new EntryUpdateTime(e.getEntryPath(), e.makePrettyPath(e.getEntryPath()), e.getEntryType(), new Timestamp(e.getLastUpdated().getTime())))
                 .filter((EntryUpdateTime entryUpdateTime) -> filter == null || filter.isBlank() || entryUpdateTime.getPath().toLowerCase().contains(filter.toLowerCase()))
                 .sorted(Comparator.comparing(EntryUpdateTime::getLastUpdateDate, Comparator.nullsLast(Comparator.reverseOrder())))
                 .collect(Collectors.toList());
 
-        // Grab subset if necessary
         if (count != null) {
-            return sortedEntries.subList(0, Math.min(count, sortedEntries.size()));
+            return filteredEntries.subList(0, Math.min(count, filteredEntries.size()));
```

Comment:
It was like this, but I think you can just limit on the stream:

```
...
.limit(count != null ? count : Integer.MAX_VALUE)
.collect(...
```
Optionally have a default value for count, which I think makes sense, and then you don't even have to do the null check.